src/conduit/api.cljs

```diff
@@ -22,17 +23,32 @@
   (str "https://conduit.productionready.io/api/" path))
 
 (defn- parse-body [res]
-  (-> (:body res)
+  (-> res
       js/JSON.parse
       (js->clj :keywordize-keys true)))
 
-(defn fetch
-  ([endpoint]
-   (fetch endpoint nil))
-  ([endpoint params]
-   (fetch endpoint params nil))
-  ([endpoint params slug]
-   (-> (->endpoint endpoint slug)
-       ->uri
-       (xhr/get {:query-params params})
-       (p/then parse-body))))
+(defn- ->json [params]
+  (.stringify js/JSON (clj->js params)))
+
+(defn- ->xhr [uri xhr-fn params]
+  (-> uri
+      (xhr-fn params)
+      (p/then (fn [{status :status body :body :as response}]
+                (condp = status
+                  status/ok (p/resolved (parse-body body))
+                  (p/rejected (parse-body body)))))))
+
+(defn fetch [{:keys [endpoint params slug method type headers]}]
+  (let [xhr-fn (case method
+                 :post xhr/post
+                 :put xhr/put
+                 :patch xhr/patch
+                 xhr/get)
+         xhr-params {:query-params (when (not (= method :post)) params)
```

Comment:
use `when-not` or `not=` instead of `(when (not (= ...`

Reply:
Oops! Didn't know the trick, it's so handy. Thanks